os/arch/arm/src/amebasmart/amebasmart_mipi.c

```diff
@@ -527,6 +524,26 @@ static int amebasmart_mipi_transfer(FAR struct mipi_dsi_host *dsi_host, FAR cons
 		rx_data_len = 0;
 	}
 	return OK;
+Fail_case:
+#ifdef CONFIG_PM
+	bsp_pm_domain_control(BSP_MIPI_DRV, 0);
+#endif
+#ifdef CONFIG_SMP
+	spin_unlock(&g_rtl8730e_config_dev_s_underflow);
+#endif
+	if (send_cmd_done == 0) {
+		sem_init(&g_send_cmd_done, 0, 0);
+	}
+	if (MIPI_LPTX_IS_READ(msg->type)) {
+		sem_init(&g_read_cmd_done, 0, 0);
```

Comment:
In this case, don't you need to check the `receive_cmd_done` like send_cmd_done?